agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/ConfigurationBuilder.java

```diff
@@ -841,7 +841,9 @@ static Configuration getConfigurationFromEnvVar(String json) {
 
     Configuration configuration = getConfiguration(json, JsonOrigin.ENV_VAR);
 
-    if (configuration.connectionString != null) {
+    // restrict connection string in APPLICATIONINSIGHTS_CONFIGURATION_CONTENT for App Service only
+    if (configuration.connectionString != null
+        && DiagnosticsHelper.useAppSvcRpIntegrationLogging()) {
```

Comment:
I don't love that this method name looks like it's related to logging, i'll send a follow-up PR to rename